src/beignet/structure/_backbone_coordinates_to_dihedrals.py

```diff
@@ -0,0 +1,106 @@
+import torch
+from torch import Tensor
+
+from beignet import dihedral
+
+
+def backbone_coordinates_to_dihedrals(
+    backbone_coordinates: Tensor,
+    mask: Tensor,
+    residue_index: Tensor | None = None,
+    chain_id: Tensor | None = None,
+) -> tuple[Tensor, Tensor]:
```

Comment:
Off-by-one error in masking, I think. You use i + 1 for omega and phi but pad in the opposite direction.

Reply:
masking is fixed